functions/slack/app/slack_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slack
+
+import (
+	"context"
+	"log"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+
+	"google.golang.org/api/kgsearch/v1"
+	"google.golang.org/api/option"
+)
+
+var projectID string
+
+// TestMain sets up the config rather than using the config file
+// which contains placeholder values.
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+	projectID = os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	if projectID == "" {
+		log.Print("GOLANG_SAMPLES_PROJECT_ID is unset. Skipping.")
+		return
+	}
+	config = &configuration{
+		ProjectID: projectID,
+		Token:     "cbdkUvSjpoiSfaHDhVNiXOs4",
+		Key:       "AIzaSyCsNvMEF7iGUM-Nu6m4ARpQw39Txzl4hJU",
```

Comment:
Are these private keys? If so, they need to be revoked immediately. Otherwise, someone will find this and use the keys.

We'll have to set up private keys separately. For now, will you add these as environment variables?

```
token := os.Getenv("GOLANG_SAMPLES_KG_TOKEN")
key := os.Getenv("GOLANG_SAMPLES_KG_KEY")
```

Then I'll see about setting those variables in Kokoro.